src/DotNetty.Buffers/UnpooledHeapByteBuffer.cs

```diff
@@ -217,6 +218,19 @@ public override long GetLong(int index)
             return this._GetLong(index);
         }
 
+        public override int GetMedium(int index)
+        {
+            this.EnsureAccessible();
+            return this._GetMedium(index).ToMediumInt();
```

Comment:
while it is not in line with other methods it simplifies conversions across the board. Could you pls unify with the Java code? It is fine for other types, but Medium does not map to framework's native type and signed<->unsigned conversion become confusing and error prone so reducing places where they're needed is a win.

---
In reply to: [114223854](https://github.com/Azure/DotNetty/pull/244#discussion_r114223854) [](ancestors = 114223854)